File: app/models/user/Language.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.user;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import models.base.VersionedModel;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

@Entity
public class Language extends VersionedModel {

    @Id
    private String code;

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Language language)) return false;
        return new EqualsBuilder().append(code, language.code).build();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37).append(code).build();
    }
}

File: app/models/user/Language.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.user

import jakarta.persistence.{Entity, Id}
import models.base.VersionedModel

import scala.compiletime.uninitialized

@Entity
class Language extends VersionedModel:
  @Id var code: String = uninitialized
  var name: String     = uninitialized

  override def equals(o: Any): Boolean = o match
    case l: Language => this.code == l.code
    case _           => false

  override def hashCode: Int = if code != null then code.hashCode else 0
